microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,587 @@
+        shell=True, 
+        sudo=True
+    )
+    if cmd_res.stdout == 'yes':
+        return True
+    else:
+        return False
+
+def found_sys_log(node: Node, message: str) -> bool:
+    if check_is_file(node, '/var/log/messages'):
+        _ret = node.tools[Cat].read_with_filter(
+            "/var/log/messages",
+            message,
+            sudo=True
+        )
+    elif check_is_file(node, '/var/log/syslog'):
+        _ret = node.tools[Cat].read_with_filter(
+            "/var/log/syslog",
+            message,
+            sudo=True
+        )
+    else:
+        _ret = node.execute(
+            f'dmesg | grep -i {message}',
+            shell=True,
+            sudo=True
+        ).stdout
+    
+    if message in _ret:
+        return True
+    else:
+        return False
+
+def is_vm_running(node: Node, log: Logger, timeout: int = 300) -> bool:
+    is_ready, tcp_error_code = wait_tcp_port_ready(
+        node.public_address,
+        node.public_port,
+        log=log,
+        timeout=timeout,
+    )
+    return is_ready    
+
+
+@TestSuiteMetadata(
+    area="power",
+    category="functional",
+    description="""
+    """
+)
+class power(TestSuite):
+    REBOOT_TIME_OUT = 300
+
+    @TestCaseMetadata(
+        description="""
+        This test can be performed in Azure and Hyper-V both. But this script only covers Azure.
+        1. Prepare swap space for hibernation
+        2. Update the grub.cfg with resume=UUID=xxxx where is from blkid swap disk
+        3. Set RTC to future time or past time
+        4. Hibernate the VM
+        5. Resume the VM
+        6. Read RTC timestamp and compare to the original value
+        """,
+        priority=4
+    )
+    def power_clock_sync_hibernate(self, environment: Environment, node: Node, log: Logger) -> None:     
+        # Prepare the swap space in the target VM
+        resource_group = environment.get_information()['resource_group_name']
+        data_disk_name = 'lisa_data_disk_0'
+        disks = node.features[Disk]
+        pattern = re.compile(r"/dev/disk/azure/scsi[1-9]/lun[0-9][0-9]?", re.M)
+        cmd_result = node.execute(
+            "ls -d /dev/disk/azure/scsi*/*", shell=True, sudo=True
+        )
+        matched = find_patterns_in_lines(cmd_result.stdout, [pattern])
+        matched_disk_array = set(matched[0])
+        disk_array: List[str] = [""] * len(matched_disk_array)
+        for disk in matched_disk_array:
+            # readlink -f /dev/disk/azure/scsi1/lun0
+            # /dev/sdc
+            cmd_result = self._node.execute(
+                f"readlink -f {disk}", shell=True, sudo=True
+            )
+            disk_array[int(disk.split("/")[-1].replace("lun", ""))] = cmd_result.stdout
+        if disk_array and disk_array[0] == data_disk_name:
+            log.info("Data disk has already been added")
+        else:
+            disks.add_data_disk(1, schema.DiskType.StandardSSDLRS, 1024)
```

Comment:
this can be set in simple_requirement, add the disk during deployment.

Reply:
simple_requirement will always use some default setting for the data disk? does that default setting fit all test cases? if not, is there any way to customize the setting, or we should use this disk.add_data_disk function

Reply:
The simple_requirement or TestCaseRequirement add disk when deploying, it's different with add_data_disk in the test case. Use TestCaseRequirement to add settings. It's complex but more flexible.